Source/PeleLM.cpp

```diff
@@ -8267,6 +8267,7 @@ PeleLM::getDiffusivity (MultiFab* diffusivity[AMREX_SPACEDIM],
 
    const int offset    = AMREX_SPACEDIM + 1;          // No diffusion coeff for vels or rho
    int       diff_comp = state_comp - offset;
+   if (state_comp == Temp) diff_comp -= 1;            // Because RhoH is squeezed in between.
```

Comment:
Well, there is no diffusivity for RhoH so I guess we should not call with this routine with state_comp == RhoH. But when I look back, this was not need before EB, so there might be another way to go.

Reply:
Yeah, maybe we should add an assert there rather than trying to program more complex logic.  the fact is that the transport coefficients do not align with the state components really.  Let's remove line 8270 and add a comment about that, but otherwise leave it.

Reply:
Not sure I understand. I do need to have a -1 if state_comp is Temp for it to work properly.

Reply:
Only if it is called specifically for Temp, and then ncomp better be 1.  If it is called for first_spec, then ncomp better be such that it does step off the end.  Those would be the asserts I meant to add.  Looking through PeleLM now anyway, the calls are ok, but only by luck (or careful planning...).

Reply:
Careful planning obviously ... Okay I've asserted that there is only two ways to call this routine, as you mentioned.